sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/SumAggregation.java

```diff
@@ -53,6 +53,20 @@ private SumAggregation() {}
     throw new IllegalArgumentException("Invalid instrument value type");
   }
 
+  @Override
+  public boolean isCompatibleWithInstrument(InstrumentDescriptor instrumentDescriptor) {
+    switch (instrumentDescriptor.getType()) {
+      case COUNTER:
+      case OBSERVABLE_COUNTER:
+      case UP_DOWN_COUNTER:
+      case OBSERVABLE_UP_DOWN_COUNTER:
+      case HISTOGRAM:
+        return true;
+      default:
+        return false;
```

Comment:
I suspect there will be people who want to `sum` their gauge values. Not sure why we need to restrict this at this point.

Reply:
Reasoning for this is same as from this [comment](https://github.com/open-telemetry/opentelemetry-java/pull/4338#discussion_r843056505). If we relax gauge, we should relax sum as well.